src/Facets/PolymerCCTPFacet.sol

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.17;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import {ILiFi} from "../Interfaces/ILiFi.sol";
+import {ITokenMessenger} from "../Interfaces/ITokenMessenger.sol";
+import {IPolymerCCTPFacet, PolymerCCTPData} from "../Interfaces/IPolymerCCTP.sol";
+import {LiFiData} from "../Helpers/LiFiData.sol";
+import {LibAsset} from "../Libraries/LibAsset.sol";
+import {LibUtil} from "../Libraries/LibUtil.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import {ILiFi} from "../Interfaces/ILiFi.sol";
+import {ITokenMessenger} from "../Interfaces/ITokenMessenger.sol";
+import {IPolymerCCTPFacet, PolymerCCTPData} from "../Interfaces/IPolymerCCTP.sol";
+import {LiFiData} from "../Helpers/LiFiData.sol";
+import {LibAsset} from "../Libraries/LibAsset.sol";
+import {LibSwap} from "../Libraries/LibSwap.sol";
+import {SwapperV2} from "../Helpers/SwapperV2.sol";
+import {Validatable} from "../Helpers/Validatable.sol";
+import {InvalidReceiver, InvalidAmount, CannotBridgeToSameNetwork, InvalidConfig} from "../Errors/GenericErrors.sol";
+import {InvalidCallData} from "../Errors/GenericErrors.sol";
+import {LibDiamond} from "../Libraries/LibDiamond.sol";
+
+/// @title PolymerCCTPFacet
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality for bridging USDC through Polymer CCTP
+/// @custom:version 1.0.0
+contract PolymerCCTPFacet is IPolymerCCTPFacet, ILiFi, ReentrancyGuard, SwapperV2, Validatable, LiFiData {
+    ITokenMessenger public immutable TOKEN_MESSENGER;
+    address public immutable USDC;
+    address payable public immutable POLYMER_FEE_RECEIVER;
+
+    // Note: we can't use the validateBridgeData because it doesn't support the NON_EVM_ADDRESS PATH
```

Comment:
we could use validateBridgeData cause it just makes sure that _receiver is non-zero which is valid for both paths. 
However, the current implementation is slightly more gas-efficient so I am fine to leave it as is.
But we should remove this comment.